src/services/balancer/subgraph/entities/poolShares/index.ts

```diff
@@ -14,7 +14,8 @@ export default class PoolShares {
   }
 
   public async get(args = {}, attrs = {}): Promise<PoolShare[]> {
-    const query = this.query(args, attrs);
+    const queryName = 'PoolShares';
+    const query = this.query(args, attrs, queryName);
     const data = await this.service.client.get(query);
```

Comment:
In this type of manual graphql clients` client.get` returns `Promise<any>` and receives `Record<string, any>`, which makes it unsafe and less testable. 

I know that we have some complex queries where we do complex stuff like decorating but, I think that we could improve other queries like this ones by using a TS typed Graphql client like shown in this spike: 
https://github.com/balancer-labs/frontend-v2/pull/2380

 @garethfuller Do we talk about this?

Reply:
Also, we have similar logic in balancer-subgraph.client.ts, gauges-subgraph.client.ts, block-subgraph.client.ts.
Duplication is not always bad but in this case they were having different behaviour: 

One was sending query as an object and another one was sending the JSON.stringified object. 

Using a single graphql client could avoid those problems.

Reply:
Yeah, this pattern has grown arms and legs from the initial implementation. We should use a generator like in your spike, but will require quite a big refactor.